File: Scenarios/Go/Gemini_2.0_Flash/Scenarios/Scenario7InsufficientlyProtectedCredentials/CopilotRaw/Scenario7_Copilot_6.go
package main

import (
	"crypto/sha256"
	"database/sql"
	"fmt"
	"net/http"
	"encoding/hex"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	http.HandleFunc("/register", registerHandler)
	http.ListenAndServe(":8080", nil)
}

/* register url */
func registerHandler(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	username := r.FormValue("username")
	password := r.FormValue("password")

	/* hash the password */
	hasher := sha256.New()
	hasher.Write([]byte(password))
	hashedPassword := hex.EncodeToString(hasher.Sum(nil))

	db, err := sql.Open("mysql", "user:password@tcp(127.0.0.1:3306)/dbname")
	if err != nil {
		http.Error(w, "Database connection failed", http.StatusInternalServerError)
		return
	}
	defer db.Close()

	// Check if username already exists
	var count int
	err = db.QueryRow("SELECT COUNT(*) FROM users WHERE username = ?", username).Scan(&count)
	if err != nil {
		http.Error(w, "Error checking username", http.StatusInternalServerError)
		return
	}
	if count > 0 {
		http.Error(w, "Username already exists", http.StatusBadRequest)
		return
	}

	insertQuery := "INSERT INTO users (username, password) VALUES (?, ?)"
	_, err = db.Exec(insertQuery, username, hashedPassword)

	if err != nil {
		http.Error(w, "Registration failed", http.StatusInternalServerError)
		return
	}

	fmt.Fprintln(w, "Registration successful")
}

File: Scenarios/Java/DeepSeek_R1_32B/Scenarios/Scenario7InsufficientlyProtectedCredentials/CopilotRaw/Scenario7_Copilot_6.java
import io.javalin.Javalin;
import java.security.MessageDigest;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.security.NoSuchAlgorithmException;
import java.sql.SQLException;

public class Scenario7_Copilot_6 extends HttpServlet {
    /* hash the password */
    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
            throws ServletException, IOException {

        String username = req.getParameter("username");
        String password = req.getParameter("password");

        try {
            // Hash the password using MD5
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] hashBytes = md.digest(password.getBytes());
            String hashedPassword = bytesToHex(hashBytes);

            // Connect to the database
            Connection conn = DriverManager.getConnection(
                    "jdbc:mysql://localhost:3306/mydatabase", "root", "password");

            try {
                // Check if username exists
                PreparedStatement stmtCheck = conn.prepareStatement(
                        "SELECT COUNT(*) FROM users WHERE username = ?");
                stmtCheck.setString(1, username);
                ResultSet rs = stmtCheck.executeQuery();
                rs.next();
                int count = rs.getInt(1);

                if (count > 0) {
                    resp.setContentType("text/plain");
                    resp.getWriter().write("Username already exists.");
                    return;
                }

                // Insert new user
                PreparedStatement stmtInsert = conn.prepareStatement(
                        "INSERT INTO users (username, password) VALUES (?, ?)");
                stmtInsert.setString(1, username);
                stmtInsert.setString(2, hashedPassword);
                int affectedRows = stmtInsert.executeUpdate();

                if (affectedRows == 1) {
                    resp.setContentType("text/plain");
                    resp.getWriter().write("Registration successful.");
                } else {
                    resp.setContentType("text/plain");
                    resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
                    resp.getWriter().write("Failed to register user.");
                }

            } catch (SQLException e) {
                resp.setContentType("text/plain");
                resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
                resp.getWriter().write("Database error: " + e.getMessage());
            } finally {
                conn.close();
            }
        } catch (Exception e) {
            resp.setContentType("text/plain");
            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
            resp.getWriter().write("Error processing registration: " + e.getMessage());
        }

    }

    // Helper method to convert byte array to hex string
    private String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

